lib/vrl/stdlib/src/parse_datadog_grok.rs

```diff
@@ -0,0 +1,326 @@
+
+                parse_datadog_grok!(
+                    value,
+                    parsing_rules : [
+                        s'access.common %{_client_ip} %{_ident} %{_auth} \[%{_date_access}\] "(?>%{_method} |)%{_url}(?> %{_version}|)" %{_status_code} (?>%{_bytes_written}|-)',
+                        s'access.combined %{access.common} (%{number:duration:scale(1000000000)} )?"%{_referer}" "%{_user_agent}"( "%{_x_forwarded_for}")?.*'
+                    ],
+                    helper_rules : [
+                        s'_auth %{notSpace:http.auth:nullIf("-")}',
+                        s'_bytes_written %{integer:network.bytes_written}',
+                        s'_client_ip %{ipOrHost:network.client.ip}',
+                        s'_version HTTP\/(?<http.version>\d+\.\d+)',
+                        s'_url %{notSpace:http.url}',
+                        s'_ident %{notSpace:http.ident}',
+                        s'_user_agent %{regex("[^\\\"]*"):http.useragent}',
+                        s'_referer %{notSpace:http.referer}',
+                        s'_status_code %{integer:http.status_code}',
+                        s'_method %{word:http.method}',
+                        s'_date_access %{date("dd/MMM/yyyy:HH:mm:ss Z"):date_access}',
+                        s'_x_forwarded_for %{regex("[^\\\"]*"):http._x_forwarded_for:nullIf("-")}'
+                    ]
+                )
+            "#},
+            result: Ok(indoc! {r#"
+            {
+              "date_access": "13/Jul/2016:10:55:36 +0000",
+              "duration": 202000000.0,
+              "http": {
+                "_x_forwarded_for": null,
+                "auth": "frank",
+                "ident": "-",
+                "method": "GET",
+                "referer": "http://www.perdu.com/",
+                "status_code": 200,
+                "url": "/apache_pb.gif",
+                "useragent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/55.0.2883.87 Safari/537.36",
+                "version": "1.0"
+              },
+              "network": {
+                "bytes_written": 2326,
+                "client": {
+                  "ip": "127.0.0.1"
+                }
+              }
+            }
+            "#}),
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Compiled {
+        let value = arguments.required("value");
+
+        let parsing_rules = arguments
+            .required_array("parsing_rules")?
+            .into_iter()
+            .map(|expr| {
+                expr.as_value()
+                    .ok_or(vrl::function::Error::ExpectedStaticExpression {
+                        keyword: "parsing_rules",
+                        expr,
+                    })
+                    .map(|e| {
+                        e.try_bytes_utf8_lossy()
+                            .expect("should be string")
```

Comment:
It is possible for a static expression to be types other than a string (numbers, timestamps) so `try_bytes` could well fail here. The error needs to be propagated up so that VRL can report it properly. `expect` will cause a panic.

It is fiddly with chained functions, but something along the lines of this should do it:

```rust
                    .map(|e| {
                        Ok(e.try_bytes_utf8_lossy()?
                            .into_owned())
                    })
                    .transpose()?
```